rrfs-test/IODA/python/bufr2ioda_msonet_prepbufr.py

```diff
@@ -0,0 +1,562 @@
+        return
+    logger.debug(f"The DATA_PATH is: {DATA_PATH}")
+
+    # ============================================
+    # Make the QuerySet for all the data we want
+    # ============================================
+    start_time = time.time()
+
+    logger.info('Making QuerySet')
+    q = bufr.QuerySet(subsets)
+
+    # ObsType
+    q.add('observationType', '*/TYP')
+
+    # MetaData
+    q.add('latitude', '*/YOB')
+    q.add('longitude', '*/XOB')
+    q.add('dataProviderOrigin', '*/PRVSTG')
+    q.add('dataProviderSubOrigin', '*/SPRVSTG')
+    q.add('stationIdentification', '*/SID')
+    q.add('stationElevation', '*/ELV')
+    q.add('timeOffset', '*/DHR')
+    q.add('pressure', '*/P___INFO/P__EVENT{1}/POB')
+    q.add('height', '*/Z___INFO/Z__EVENT{1}/ZOB')
+
+    # ObsValue
+    q.add('stationPressure', '*/P___INFO/P__EVENT{1}/POB')
+    q.add('airTemperature', '*/T___INFO/T__EVENT{1}/TOB')
+    q.add('dewPointTemperature', '*/Q___INFO/TDO')
+    q.add('virtualTemperature', '*/T___INFO/TVO')
+    q.add('specificHumidity', '*/Q___INFO/Q__EVENT{1}/QOB')
+    q.add('windEastward', '*/W___INFO/W__EVENT{1}/UOB')
+    q.add('windNorthward', '*/W___INFO/W__EVENT{1}/VOB')
+
+    # QualityMark
+    q.add('pressureQM', '*/P___INFO/P__EVENT{1}/PQM')
+    q.add('stationPressureQM', '*/P___INFO/P__EVENT{1}/PQM')
+    q.add('heightQM', '*/Z___INFO/Z__EVENT{1}/ZQM')
+    q.add('airTemperatureQM', '*/T___INFO/T__EVENT{1}/TQM')
+    q.add('specificHumidityQM', '*/Q___INFO/Q__EVENT{1}/QQM')
+    q.add('windEastwardQM', '*/W___INFO/W__EVENT{1}/WQM')
+    q.add('windNorthwardQM', '*/W___INFO/W__EVENT{1}/WQM')
+
+    # ObsError
+    q.add('pressureOE', '*/P___INFO/P__BACKG/POE')
+    q.add('airTemperatureOE', '*/T___INFO/T__BACKG/TOE')
+    q.add('specificHumidityOE', '*/Q___INFO/Q__BACKG/QOE')
+    q.add('windEastwardOE', '*/W___INFO/W__BACKG/WOE')
+    q.add('windNorthwardOE', '*/W___INFO/W__BACKG/WOE')
+
+    end_time = time.time()
+    running_time = end_time - start_time
+    logger.debug(f'Running time for making QuerySet : {running_time} seconds')
+
+    # ==============================================================
+    # Open the BUFR file and execute the QuerySet to get ResultSet
+    # Use the ResultSet returned to get numpy arrays of the data
+    # ==============================================================
+    start_time = time.time()
+
+    logger.info('Executing QuerySet to get ResultSet')
+    with bufr.File(DATA_PATH) as f:
+        try:
+            r = f.execute(q)
+        except Exception as err:
+            logger.info(f'Return with {err}')
+            return
+
+    # ObsType
+    logger.debug(" ... Executing QuerySet for MSONET: get ObsType ...")
+    obstyp = r.get('observationType', type='int32')
+    logger.info('Executing QuerySet: get metadata')
+
+    # MetaData
+    lat = r.get('latitude')
+    lon = r.get('longitude')
+    lon[lon > 180] -= 360  # Convert Longitude from [0,360] to [-180,180]
```

Comment:
Let's comment this line to keep it in 0,360. That is what we are using currently for yaml. And this was leading to no observations assimilated... at least for me. Make sure to update the valid range too.

Reply:
@delippi done, commented the conversion line, if needed to switch back later.